### PR TITLE
Allow resuming upload without local checksum calculation

### PR DESCRIPTION
**Pull request by danvalencia**
Local checksum calculation is problematic and causes crashes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
 node_modules
 npm-debug.log
 .DS_Store
+.idea/
+.npmrc
```
Comment (heryxpc) [inline .gitignore:5]: .npmrc is used on `folio-web` to save Nexus repository URL and credentials placeholder.

File: src/Upload.js [hunk 2]
```diff
@@ -69,25 +70,35 @@ export default class Upload {
...
 
-      this.debug('Validation passed, resuming upload')
-      await processor.run(uploadChunk, resumeIndex)
+      this.debug(` - Remote index: ${remoteResumeIndex}`)
+
+      if (opts.validateChecksum) {
+        try {
+          await processor.run(validateChunk, 0, resumeIndex)
+        } catch (e) {
+          this.debug('Validation failed, starting from scratch')
```
Comment (heryxpc) [inline src/Upload.js:86]: Is there a way to notify the user of this error?

**Review by heryxpc — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (danvalencia) [reply .gitignore:5]: Yes, that's how I'm using it locally, but we don't want credentials stored in git, hence why it should be added to .gitignore.

File: src/Upload.js (hunk 2 above)
Comment (danvalencia) [reply src/Upload.js:86]: I don't think there's a need for it. This is the default behavior of the library, if the validation fails it simply starts the process from scratch. But with this PR I'm bypassing this validation, as it breaks with large files, and I don't think we need it.

File: .gitignore (hunk 1 above)
Comment (heryxpc) [reply .gitignore:5]: This is what we have on our git repositories of other JS code:
```
registry=https://repo.helix.re/repository/helix-npm/
always-auth=true
_auth=${NEXUS_AUTH}
email=${NEXUS_EMAIL}
```
I'm not sure if it is used by our CI pipelines to build the code, but that is what we have.
If there is no use of this info for the project, I believe it is ok to add it to `.gitignore`.

File: src/Upload.js (hunk 2 above)
Comment (heryxpc) [reply src/Upload.js:86]: Got it 👍

File: .gitignore (hunk 1 above)
Comment (mrlubos) [reply .gitignore:5]: Yes @heryxpc, `.npmrc` is safe to push to Git as no credentials are stored in the file (unless you don't want the registry path to be visible publicly).

**Pull request merged**